### PR TITLE
Fix custom_filed KeyError traceback

### PR DESCRIPTION
**Pull request by Kani999**

### CODE REVIEW

**Comment by Kani999** [comment]
- Closes #457

File: pynetbox/core/response.py [hunk 1]
```diff
@@ -58,9 +58,7 @@ def get_return(lookup, return_fields=None):
 
 
 def flatten_custom(custom_dict):
-    return {
-        k: v if not isinstance(v, dict) else v["value"] for k, v in custom_dict.items()
-    }
+    return {k: v for k, v in custom_dict.items()}
```
Comment (jcollie) [inline pynetbox/core/response.py:61]: This really seems like a NOOP to me. Perhaps it would be better to eliminate calls to flatten_custom altogether?

**Comment by srfwx** [comment]
We need it for Objects and Multiple Objects custom fields (although it needs to be adapted too)

**Comment by Kani999** [comment]
Okay, so it should return something like.

**Object**: `inventory-item`
**Custom Fileds**:
- `test_json_custom_field`  `{'x': 1, 'y': 2}` - JSON
- `test_list_custom_field` `['x', 'y']` - MultipleSelection
- `test_multiple_objects_custom_field` [Dev1, Dev2] - MultipleObjects (Device) 
- `abra_discovered`  `False` - Boolean 
- `asset_numbers` `T5060` - String
- `inventarizace` `{'x': 1, 'y': 2}` - JSON
`inventory_monitor_last_probe` `2022-11-23` - Date
`inventory_monitor_probe` - `Probe` - Object (from plugin) 

**Response**
```JSON
{
   "id":5023,
   "url":"https://NETBOX-INSTANCE/api/dcim/inventory-items/5023/",
   "display":"0/6/1",
   "device":61,
   "parent":"None",
   "name":"0/6/1",
   "label":"",
   "role":5,
   "manufacturer":"None",
   "part_id":"",
   "asset_tag":"None",
   "discovered":true,
   "description":"",
   "component_type":"None",
   "component_id":"None",
   "component":"None",
   "tags":[
      
   ],
   "custom_fields":{
      "test_json_custom_field":{
         "x":1,
         "y":2
      },
      "test_list_custom_field":[
         "x",
         "y"
      ],
      "test_multiple_objects_custom_field":[
         {
            "id":1,
            "url":"https://NETBOX-INSTANCE/api/dcim/devices/1/",
            "display":"DEV1",
            "name":"DEV1"
         },
         {
            "id":119,
            "url":"https://NETBOX-INSTANCE/api/dcim/devices/119/",
            "display":"DEV2",
            "name":"DEV2"
         },
      ],
      "abra_discovered":false,
      "asset_numbers":"T5060",
      "inventarizace":{
         "x":1,
         "y":2
      },
      "inventory_monitor_active":true,
      "inventory_monitor_last_probe":"2022-11-23",
      "inventory_monitor_probe":{
         "id":1648,
         "url":"https://NETBOX-INSTANCE/api/plugins/inventory-monitor/probes/1648/",
         "display":"Object",
         "name":"0/6/1",
         "time":"2022-11-23T01:16:47.492262Z"
      },
      "inventory_monitor_probed":true
   },
   "created":"2022-09-06T10:03:36.248425Z",
   "last_updated":"2022-11-23T08:46:57.907296Z",
   "_depth":0
}
```

File: pynetbox/core/response.py (hunk 1 above)
Comment (FliesLikeABrick) [reply pynetbox/core/response.py]: @srfwx  weighed in on #457 back in June providing reasoning that `flatten_custom` is needed, and it looks like some discussion around this is the main thing preventing merge of an appropriate fix.

**Comment by michael-salo** [comment]
Is there any chance this pull request can be reviewed and merged into an upstream version of pynetbox? We're hitting the same bug when trying to create netbox_prefix via ansible with custom_fields.

**Comment by arthanson** [comment]
Will take a look into it, probably Wed or Thursday before I can get to it.

**Pull request merged**